web/src/OverviewTableColumns.tsx

```diff
@@ -209,6 +209,89 @@ function statusSortKey(row: RowValues): string {
   return `${order}${row.name}`
 }
 
+/**
+ * Header components
+ */
+export function ResourceSelectionHeader({
+  rows,
+  column,
+}: HeaderProps<RowValues>) {
+  const { selected, isSelected, select, deselect } = useResourceSelection()
+
+  const selectableResourcesInTable = useMemo(() => {
+    const resources: string[] = []
+    rows.forEach(({ original }) => {
+      if (original.selectable) {
+        resources.push(original.name)
+      }
+    })
+
+    return resources
+  }, [rows])
+
+  function getSelectionState(resourcesInTable: string[]): {
+    indeterminate: boolean
+    checked: boolean
+  } {
+    let selectionCount = 0
+
+    for (let i = 0; i < resourcesInTable.length; i++) {
+      const selected = isSelected(resourcesInTable[i])
+      if (selected) {
+        selectionCount++
+      }
+
+      // If at least one item has been selected, but the
+      // current item isn't, we can exit the loop early
+      if (!selected && selectionCount > 0) {
```

Comment:
As long as we're doing small optimizations, this stops early for [selected, unselected], but not [unselected, selected] :)

totally fine as-is, but I think this would be easier to understand without counting, e.g.:
```
let anySelected = false
let anyUnselected = false
resourcesInTable.forEach(r => {
  if (isSelected(r)) {
    anySelected = true
  } else {
    anyUnselected = true
  }

  if (anySelected && anyUnselected) {
    break
  }
}

return {
  indeterminate: anySelected !== anyUnselected
  checked: !anyUnselected
}
```

Reply:
ah, great suggestion! i'll update.